tests/pipeline/test_transform_cis_soc_data.py

```diff
@@ -7,22 +7,45 @@ def test_transform_cis_soc_data(spark_session):
     input_df = spark_session.createDataFrame(
         data=[
             ("JOB TITLE 1", "JOB1", 5),
-            ("JOB TITLE 1", "JOB1", 5),
-            ("JOB TITLE 2", "JOB2", 6),
+            ("JOB TITLE 1", "JOB1", 5),  # Complete duplicates get deduplicated
+            ("JOB TITLE 2", "JOB2", 6),  # Less specific, so this is dropped
             ("JOB TITLE 2", "JOB2", 66),
-            ("JOB TITLE 2", "JOB2", 77),
-            ("JOB TITLE 2", "JOB2", "un"),
+            ("JOB TITLE 2", "JOB2", 77),  # Two codes with equally high specificity, so require manual resolution
```

Comment:
The test is looking correct apart from "JOB2". Following the expected behaviour:

1. "un" is dropped, because other records have codes
2. 6 is dropped, because other records have more specific codes
3. 77 and 66 are put into the conflicted df, because we can't automatically resolve the conflict between them
4. No records for "JOB2" get included in the `output_df`

We wouldn't expect to see the "un" or 6 code in any of the returned df, because our automatic resolution knows that there are preferred records. The 77 is currently incorrectly being chosen as the correct outcome. I think this is because they are ordered alphanumerically, instead of being ordered by code length specifically